### PR TITLE
Review and linting of wasm and runtime modules

### PR DESCRIPTION
**Pull request by andreaTP**

### CODE REVIEW

File: runtime/src/main/java/com/dylibso/chicory/runtime/Instance.java [hunk 1]
```diff
@@ -95,18 +96,17 @@ public Instance(
...
         this.machine = machineFactory.apply(this);
         this.roughTables = tables.clone();
+        this.tables = new TableInstance[this.roughTables.length];
         this.elements = elements.clone();
         this.exports = exports;
-        this.start = start;
         this.listener = listener;
 
         if (initialize) {
-            initialize(this.start);
```
[the remark this answers is not in the document]
Comment (andreaTP) [reply runtime/src/main/java/com/dylibso/chicory/runtime/Instance.java:104]: So far, we don't have any strict rule about the usage of `this`.
If you want to enforce a specific style, the best would be to have a linting rule in spotless to make sure it's consistent across the codebase.

In this specific case, although, it doesn't apply as `start` is not a class filed anymore, but just a local variable: https://github.com/dylibso/chicory/pull/481/files#diff-5b264b6d433604465d048b58b2aded7984824cd861b6bd519b63698e8d171e81L62

[the remark this answers is not in the document]
Comment (evacchi) [reply runtime/src/main/java/com/dylibso/chicory/runtime/Instance.java:104]: Oooh that's why 👍

File: runtime/src/main/java/com/dylibso/chicory/runtime/Instance.java [hunk 2]
```diff
@@ -656,38 +655,33 @@ public Instance build() {
...
-                        break;
+                if (imprt.importType() == FUNCTION) {
+                    var type = ((FunctionImport) imprt).typeIndex();
+                    if (type >= this.module.typeSection().typeCount()) {
+                        throw new InvalidException("unknown type");
+                    }
+                    functionTypes[funcIdx] = type;
+                    // The global function id increases on this table
+                    // function ids are assigned on imports first
+                    imports[importId++] = imprt;
```
Comment (electrum) [inline runtime/src/main/java/com/dylibso/chicory/runtime/Instance.java:666]: This could be moved below and the else block removed

**Review by electrum — APPROVED** [review]

File: runtime/src/main/java/com/dylibso/chicory/runtime/Instance.java (hunk 2 above)
Comment (andreaTP) [reply runtime/src/main/java/com/dylibso/chicory/runtime/Instance.java:666]: right :+1:

**Pull request merged**